test/e2e_env/multizone/localityawarelb/locality_aware_multizone_hybrid.go

```diff
@@ -0,0 +1,254 @@
+			)).
+			Install(TestServerUniversal("test-server-az-1", mesh, WithAdditionalTags(
+				map[string]string{
+					"k8s.io/az": "az-1",
+				}),
+				WithServiceName("test-server_locality-aware-lb_svc_80"),
+				WithArgs([]string{"echo", "--instance", "test-server-az-1-zone-4"}),
+			)).
+			Install(TestServerUniversal("test-server-node-2", mesh, WithAdditionalTags(
+				map[string]string{
+					"k8s.io/node": "node-2",
+				}),
+				WithServiceName("test-server_locality-aware-lb_svc_80"),
+				WithArgs([]string{"echo", "--instance", "test-server-node-2-zone-4"}),
+			)).
+			Install(TestServerUniversal("test-server-no-tags", mesh,
+				WithArgs([]string{"echo", "--instance", "test-server-no-tags-zone-4"}),
+				WithServiceName("test-server_locality-aware-lb_svc_80"),
+			)).
+			Setup(multizone.UniZone1),
+		).To(Succeed())
+
+		// Universal Zone 5
+		Expect(NewClusterSetup().
+			Install(DemoClientUniversal(
+				"demo-client_locality-aware-lb_svc",
+				mesh,
+				WithTransparentProxy(true),
+			)).
+			Install(TestServerUniversal("test-server", mesh,
+				WithServiceName("test-server_locality-aware-lb_svc_80"),
+				WithArgs([]string{"echo", "--instance", "test-server-zone-5"}),
+			)).
+			Setup(multizone.UniZone2),
+		).To(Succeed())
+
+		// Kubernetes Zone 1
+		Expect(NewClusterSetup().
+			Install(NamespaceWithSidecarInjection(namespace)).
+			Install(democlient.Install(democlient.WithMesh(mesh), democlient.WithNamespace(namespace))).
+			Install(testserver.Install(
+				testserver.WithName("test-server"),
+				testserver.WithMesh(mesh),
+				testserver.WithNamespace(namespace),
+				testserver.WithEchoArgs("echo", "--instance", "test-server-zone-1"),
+			)).
+			Setup(multizone.KubeZone1)).ToNot(HaveOccurred())
+
+		// Kubernetes Zone 2
+		Expect(NewClusterSetup().
+			Install(NamespaceWithSidecarInjection(namespace)).
+			Install(democlient.Install(democlient.WithMesh(mesh), democlient.WithNamespace(namespace))).
+			Install(testserver.Install(
+				testserver.WithName("test-server"),
+				testserver.WithMesh(mesh),
+				testserver.WithNamespace(namespace),
+				testserver.WithEchoArgs("echo", "--instance", "test-server-zone-2"),
+			)).
+			Setup(multizone.KubeZone2)).ToNot(HaveOccurred())
+	})
+
+	E2EAfterAll(func() {
+		Expect(multizone.UniZone1.DeleteMeshApps(mesh)).To(Succeed())
+		Expect(multizone.UniZone2.DeleteMeshApps(mesh)).To(Succeed())
+		Expect(multizone.KubeZone1.TriggerDeleteNamespace(namespace)).To(Succeed())
+		Expect(multizone.KubeZone2.TriggerDeleteNamespace(namespace)).To(Succeed())
+		Expect(multizone.Global.DeleteMesh(mesh)).To(Succeed())
+	})
+
+	It("should route based on defined strategy", func() {
+		Eventually(func() (map[string]int, error) {
+			return client.CollectResponsesByInstance(multizone.UniZone1, "demo-client_locality-aware-lb_svc", "test-server_locality-aware-lb_svc_80.mesh", client.WithNumberOfRequests(100))
+		}, "30s", "500ms").Should(
```

Comment:
I think 500ms with 100 requests is a bit aggressive. It's 200rps. I'd change it to something like 2s? 5s maybe even.